sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_base_test.hpp

```diff
@@ -43,13 +45,80 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Keys { nam
         Azure::Response<T>& response,
         Azure::Core::Http::HttpStatusCode expectedCode = Azure::Core::Http::HttpStatusCode::Ok)
     {
-      auto rawResponse = response.ExtractRawResponse();
+      auto& rawResponse = response.GetRawResponse();
       EXPECT_EQ(
           static_cast<typename std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
-              rawResponse->GetStatusCode()),
+              rawResponse.GetStatusCode()),
           static_cast<typename std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
               expectedCode));
     }
+
+    static inline std::string GetUniqueName() { return Azure::Core::Uuid::CreateUuid().ToString(); }
+
+    static inline void EmptyTrashFromVault(KeyClient const& keyClient)
+    {
+      std::vector<DeletedKey> deletedKeys;
+      GetDeletedKeysSinglePageOptions options;
+      while (true)
+      {
+        auto keyResponse = keyClient.GetDeletedKeysSinglePage(options);
+        for (auto& key : keyResponse->Items)
+        {
+          deletedKeys.emplace_back(key);
+        }
+        if (!keyResponse->ContinuationToken)
+        {
+          break;
+        }
+        options.ContinuationToken = keyResponse->ContinuationToken;
+      }
+      if (deletedKeys.size() > 0)
+      {
+        for (auto& deletedKey : deletedKeys)
+        {
+          keyClient.PurgeDeletedKey(deletedKey.Name());
+        }
+        // Wait for purge is completed
+        std::this_thread::sleep_for(std::chrono::minutes(1));
+      }
+    }
+
+    static inline void RemoveAllKeysFromVault(KeyClient const& keyClient, bool waitForPurge = true)
+    {
+      std::vector<DeleteKeyOperation> deletedKeys;
+      GetPropertiesOfKeysSinglePageOptions options;
+      while (true)
+      {
+        auto keyResponse = keyClient.GetPropertiesOfKeysSinglePage(options);
+        for (auto& key : keyResponse->Items)
+        {
+          deletedKeys.emplace_back(keyClient.StartDeleteKey(key.Name));
+        }
+        if (!keyResponse->ContinuationToken)
+        {
+          break;
+        }
+        options.ContinuationToken = keyResponse->ContinuationToken;
+      }
+      if (deletedKeys.size() > 0)
+      {
+        std::cout << std::endl
+                  << "Cleaning vault. " << deletedKeys.size()
+                  << " Will be deleted and purged now...";
+        for (auto& deletedKey : deletedKeys)
+        {
+          auto readyToPurgeKey = deletedKey.PollUntilDone(std::chrono::milliseconds(1000));
+          keyClient.PurgeDeletedKey(readyToPurgeKey->Name());
+          std::cout << std::endl << "Deleted and purged key: " + readyToPurgeKey->Name();
+        }
+        std::cout << std::endl << "Complete purge operation.";
+        // Wait for purge is completed
+        if (waitForPurge)
+        {
+          std::this_thread::sleep_for(std::chrono::minutes(1));
```

Comment:
Should purge be an operation which we can poll for done?  Waiting for a minute when its already completed is wasteful

Reply:
The LRO from keyvault keys are simulated.
- DeleteKey -> We just wait until the key can be fetched by sending GET /keys/deletedKeys/{keyName}
- RecoverKey -> We wait for the key can be fetched from GET /keys/{keyName}

For Purge, there's nothing we can try to hook for identifying when the service has finished the operation.  As per @heaths  advise, I am setting some waiting time as a way to test it